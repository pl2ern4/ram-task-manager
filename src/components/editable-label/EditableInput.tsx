import { useState } from "react";
import CloseIcon from '@mui/icons-material/Close';
import CheckIcon from '@mui/icons-material/Check';
import { IconButton } from "@mui/material";
import {
    EditableInputLabel,
    EditableInputWrapper,
    EditableInputReadOnlyLabel,
    CustomTextField,
    EditableForm,
    ActionButtonWrapper
} from './styles';

interface Props {
    label: string|undefined;
    type: string;
    text: string | number | undefined;
    onChange?: (value:any)=>any
}

const EditableInput = (props:Props) => {
    const { label, type, text, onChange } = props;
    const [isEditable, setIsEditable] = useState(false);
    const [value, setValue] = useState(text);
    const [fieldValue, setFieldValue] = useState();
    const handleSaveClick = () => { 
        setValue(fieldValue);
        setIsEditable(false);
        onChange && onChange(fieldValue);
    }
    const handleTextFieldChange = (params: any) => {
        setFieldValue(params.target.value)
    }
    const handleCloseClick = () =>{
        setIsEditable(false);
    }

    let field = <EditableForm>
        <CustomTextField type={type} onChange={handleTextFieldChange} value={fieldValue} autoFocus={true}/>
        <ActionButtonWrapper>
            <IconButton aria-label="cancel" color="success" onClick={handleSaveClick}>
                <CheckIcon />
            </IconButton>  
            <IconButton aria-label="save" color="error" onClick={handleCloseClick}>
                <CloseIcon />
            </IconButton>
        </ActionButtonWrapper>
        
    </EditableForm>;
    if (!isEditable) {
        field =(<EditableInputReadOnlyLabel onClick={() => setIsEditable(true)}>
            {value}
        </EditableInputReadOnlyLabel>)
    }
    return (
        <EditableInputWrapper>
            <EditableInputLabel>
                {label}
            </EditableInputLabel>
            {field}
        </EditableInputWrapper>
    )

}

export default EditableInput;