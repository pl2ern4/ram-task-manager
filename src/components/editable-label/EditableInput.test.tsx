import { render } from '@testing-library/react';
import EditableInput from './EditableInput';

describe('EditableInput', () => {
    it('should Match Snapshot', async () => {
        const { container } = render(<EditableInput label="Test" text="test" type="number"/>)
        expect(container).toMatchSnapshot();
    });  
})