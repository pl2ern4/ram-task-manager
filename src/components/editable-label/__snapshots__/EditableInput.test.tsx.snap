// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditableInput should Match Snapshot 1`] = `
<div>
  <div
    class="css-psohlv"
  >
    <label
      class="css-eie7o"
    >
      Test
    </label>
    <label
      class="css-17045cg"
    >
      test
    </label>
  </div>
</div>
`;
